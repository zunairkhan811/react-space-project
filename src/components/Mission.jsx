import { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { fetchMissions } from '../redux/mission/missionSlice';

const Mission = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchMissions());
  }, []);

  return (
    <div className="all-missions">
      <h2>Missions</h2>
    </div>
  );
};

export default Mission;
