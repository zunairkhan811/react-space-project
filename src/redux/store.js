/* eslint-disable import/prefer-default-export */
import { configureStore } from '@reduxjs/toolkit';
import MissionReducer from './mission/missionSlice';
import rocketsReducer from './rocket/rocketSlice';

export const store = configureStore({
  reducer: {
    missions: MissionReducer,
    rockets: rocketsReducer,
  },
});
