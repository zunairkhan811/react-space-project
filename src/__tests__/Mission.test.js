import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Mission from '../components/Mission';
import { store } from '../redux/store';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Provider store={store}>
          <Mission />
        </Provider>

      </BrowserRouter>,

    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
