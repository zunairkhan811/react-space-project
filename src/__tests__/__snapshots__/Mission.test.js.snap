// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="all-missions"
>
  <table
    className="missions-table"
  >
    <thead
      className="table-header"
    >
      <tr>
        <th>
          Mission
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
        <th
          className="dot"
        >
          .
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
